src/homebase/js.cljs

```diff
@@ -167,12 +167,123 @@
          (dissoc tx :db/id))
     [tx]))
 
+
+(defn humanize-transact-error [error]
+  (condp re-find (.-message error)
+    #"\[object Object\] is not ISeqable" 
+    "Expected an array of transactions. 
+\nFor example:  transact([
+                {todo: {name: 1}}, 
+                {todo: {name: 2}}
+              ])
+"
+    
+    #"Unknown operation at \[nil nil nil nil\], expected"
+    "Expected 'retractEntity'. 
+\nFor example:  transact([['retractEntity', id]])
+"
+    
+    #"Can't use tempid in '\[:db\.fn/retractEntity"
+    "Expected a numerical id. 
+\nFor example:  transact([['retractEntity', 123]])
+"
+    
+    #"Expected number or lookup ref for entity id, got nil"
+    "Expected a numerical id. 
+\nFor example:  transact([['retractEntity', 123]])
+"
+    (.-message error)))
+
 (defn transact! [conn txs]
-  (let [txs (mapcat (comp nil->retract js->tx) txs)]
-    (d/transact! conn txs)))
+  (try 
+    (d/transact! conn (mapcat (comp nil->retract js->tx) txs))
+    (catch js/Error e 
+      (throw (js/Error. (humanize-transact-error e))))))
+(comment
```

Comment:
@rubinovitz you might already know this, but if you ever want to work on testing these comments are a good place to start. I use inline comments while developing in the repl. Then when it all works I like to cut my comments out and paste them into my test suite to use as regression tests. The code in these comments needs to be massaged into boolean assertions, but otherwise they should represent a pretty solid amount of coverage.